lib/shared/models/selection-state.js

```diff
@@ -82,19 +106,34 @@ exports.selectImage = (image) => {
 }
 
 /**
- * @summary Get drive
+ * @summary Get all selected drives' devices
+ * @function
+ * @public
+ *
+ * @returns {String[]} selected drives' devices
+ *
+ * @example
+ * for (driveDevice of selectionState.getSelectedDevices()) {
+ *   console.log(driveDevice)
```

Comment:
One specifies what it returns (drive device field) and the other specifies the state it affects (drives selected), but I guess we could clarify by renaming it `deselectOtherDrivesByDevice` although that is a bit verbose and I think the `@param` field documents what it selects a drive by quite well already.

Reply:
Maybe I'm just being a bit dim in grokking this code as I've not worked on it myself, but perhaps verbosity is a good thing? \*shrug\*

Reply:
I've improved the docs here now

Reply:
I _think_ I'm starting to finally understand it all now - and the extra comments help, thank you :)

So we have a `drive` which is an Object, and a `driveDevice` which is a String, and each `drive` Object has a `device` String property (amongst others) which corresponds to the `driveDevice` strings. And "availableDrives" (in the `store`) is a list of these `drive` Objects, and "selectedDevices" (also in the `store`) is a list of `driveDevice` strings. And there's various methods to manipulate/translate these, e.g. getting the `drive` Object(s) that correspond to a particular set of `driveDevice` strings, and vice-versa.

Why is the "availableDrives" stored as an array and not as an object mapping `driveDevice` String -> `drive` Object? (seems like that'd make a lot of the manipulations / translations a bit easier?)

And listing the drive/device-related functions in `selection-state.js` (using a modified notation) we have:
```
selectDrive(String driveDevice)
toggleDrive(String driveDevice)
deselectOtherDrives(String driveDevice)
String[] driveDevices getSelectedDevices()
Object drive getCurrentDrive()
Boolean hasDrive()
deselectDrive(String driveDevice)
deselectAllDrives()
Boolean isCurrentDrive(String driveDevice)
Boolean isDriveSelected(String driveDevice)
```
...so I think the reason I was getting so confused is that we have lots of functions named `xxxDrive` which actually take a `driveDevice` as a parameter, another function named `xxxDrive()` (`getCurrentDrive`) that returns a `drive` Object, and only one function-name named `xxxDevice` (`getSelectedDevices`) that returns a list of `driveDevice`s. :upside_down_face:  (and there's also the asymmetry of `getCurrentDrive` and `isCurrentDrive` returning / taking different types, even though they're named so similarly!)


And again it's something well outside the scope of this particular PR, but I wonder if (in future) it might make sense to split `selection-state.js` up into `drive-selection-state.js` and `image-selection-state.js` ?

Reply:
Using a dictionary/map/object instead of a list is a good idea as you should never have two identical mount points – I assume it's faster too. @jhermsmeier what do you think?

As for splitting it, that's definitely something I think we'll go towards once we get drives and HTTP as sources.

Reply:
And yeah, you've got it right, here :)

Reply:
@Shou yeah, I think it's a good idea – that will very much simplify things I reckon; performance wise the impact is negligible compared to everything else going on, as this code isn't called very often (i.e. not 1000+ op/s)

Reply:
Do you wanna do that in a follow up PR? This one has become quite the one to navigate :D